pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1236,8 +1236,26 @@ type AzureAD struct {
 	// +optional
 	Cloud *string `json:"cloud,omitempty"`
 	// ManagedIdentity defines the Azure User-assigned Managed identity.
-	// +required
+	// Cannot be set at the same time as oauth.
+	// +optional
 	ManagedIdentity ManagedIdentity `json:"managedIdentity"`
+	// OAuth defines the oauth config that is being used to authenticate.
+	// Cannot be set at the same time as managedIdentity.
+	// +optional
+	OAuth *AzureOAuth `json:"oauth,omitempty"`
+}
+
+// AzureOAuth is used to store azure oauth config values.
+// +k8s:openapi-gen=true
+type AzureOAuth struct {
+	// ClientID is the clientId of the azure active directory application that is being used to authenticate.
+	ClientID string `json:"clientId,omitempty"`
+
+	// ClientSecret is the clientSecret of the azure active directory application that is being used to authenticate.
+	ClientSecret *v1.SecretKeySelector `json:"clientSecret,omitempty"`
```

Comment:
I might be wrong but I guess if you choose OAuth you have to provide a ClientSecret right?
If so it can't be nil so you can remove the `*`